static/js/components/RunSummary.jsx

```diff
@@ -0,0 +1,423 @@
+  };
+
+  const ispending = assignment.status === "pending";
+  const isdone = assignment.status === "complete";
+  const isnotdone = assignment.status === "not observed";
+
+  const completeAction = () => {
+    handleClose();
+    return dispatch(
+      SourceAction.editAssignment({ status: "complete" }, assignment.id)
+    );
+  };
+
+  const pendingAction = () => {
+    handleClose();
+    return dispatch(
+      SourceAction.editAssignment({ status: "pending" }, assignment.id)
+    );
+  };
+
+  const notObservedAction = () => {
+    handleClose();
+    return dispatch(
+      SourceAction.editAssignment({ status: "not observed" }, assignment.id)
+    );
+  };
+
+  return (
+    <div>
+      <IconButton
+        aria-controls="simple-menu"
+        aria-haspopup="true"
+        onClick={handleClick}
+        variant="contained"
+      >
+        <BuildIcon />
+      </IconButton>
+      <Menu
+        id="simple-menu"
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        {
+          (ispending || isnotdone) && (
+          <MenuItem onClick={completeAction} variant="contained" key={`${assignment.id}_done`}>
+            Mark Observed
+          </MenuItem>
+          )
+}
+        {
+          (ispending || isdone) && (
+          <MenuItem onClick={notObservedAction} variant="contained" key={`${assignment.id}_notdone`}>
+            Mark Not Observed
+          </MenuItem>
+          )
+}
+        {
+          (isdone || isnotdone) && (
+          <MenuItem onClick={pendingAction} variant="contained" key={`${assignment.id}_pending`}>
+            Mark Pending
+          </MenuItem>
+          )
+}
+        {isdone && (
+        <MenuItem key={`${assignment.id}_upload_spec`} onClick={handleClose}>
+          Upload Spectrum
+        </MenuItem>
+        )}
+        {
+          isdone && (
+          <MenuItem
+            key={`${assignment.id}_upload_phot`}
+            variant="contained"
+            onClick={handleClose}
+          >
+            <Link
+              href={`/upload_photometry/${assignment.obj.id}`}
+              underline="none"
+              color="textPrimary"
+            >
+              Upload Photometry
+            </Link>
+          </MenuItem>
+          )
+}
+      </Menu>
+    </div>
+  );
+};
+
+SimpleMenu.propTypes = {
+  assignment: PropTypes.shape({
+    status: PropTypes.string,
+    id: PropTypes.number,
+    obj: PropTypes.shape(
+      {
+        id: PropTypes.string
+      }
+    ).isRequired
+  }).isRequired
+};
+
+
+const RunSummary = ({ route }) => {
+  const dispatch = useDispatch();
+  const observingRun = useSelector((state) => state.observingRun);
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const groups = useSelector((state) => state.groups.all);
+
+
+  // Load the observing run and its assignments if needed
```

Comment:
This comment makes it sound like we're checking whether it's already loaded and only fetching if not, which we're not doing. Should that check be in the hook below?